web/src/main/java/com/google/accompanist/web/WebView.kt

```diff
@@ -220,6 +230,41 @@ class WebViewState(webContent: WebContent) {
     val errorsForCurrentRequest = mutableStateListOf<WebViewError>()
 }
 
+class WebViewNavigator(private val coroutineScope: CoroutineScope) {
+
+    private enum class NavigationEvent { BACK, FORWARD }
+    private val navigationEvents: MutableSharedFlow<NavigationEvent> = MutableSharedFlow()
```

Comment:
I don't think I am fully understanding the need for a shared flow. Are you worried about calls from multiple threads?

Reply:
It was mainly used as it's the simplest way to obtain a mutable flow instance that has the behavioural properties I needed (it just has an additional property, the "shared" part, that's unused). It could be done with Channels instead but it just seemed simpler. I can look into making that change.

Reply:
Sorry, I meant why not just call the webview functions directly, why pipe through a flow/channel?

Reply:
I wasn't really sure that exposing the `WebView` itself to the navigator was the right thing to do, another option could be providing the navigator with references to the two methods on the `WebView`, but that also seemed like it would be a bit dangerous? The flow was something I went with in order to send instructions to the Composable in a decoupled way, without leaking references out from the Composable. I sort of looked at this as following the same sort of model as `MutableInteractionSource`.

Reply:
Cool, I understand it now. That decoupling does sound good. Let me go read how interaction source works so I can give you better feedback

Reply:
Awesome, I'll work on cleaning it up, adding tests and docs. I'll probably also look at extending it slightly to include the "navigate back/forward by X steps" functions on WebView